pkg/kubelet/volumes.go

```diff
@@ -77,12 +77,18 @@ func (kl *Kubelet) mountExternalVolumes(pod *api.BoundPod) (volumeMap, error) {
 	for i := range pod.Spec.Volumes {
 		volSpec := &pod.Spec.Volumes[i]
 
+		podRef, err := api.GetReference(pod)
```

Comment:
@markturansky It's not an etcd thing. It's that the object didn't come from apiserver, which fills in the self links.

@pmorie IMO I think for resiliency we should make GetReference fail gracefully (return a mostly-complete ObjectReference) if it can't parse the self link.